### DISCUSSION TITLE
Re-run dependabot workflows as another actor

### DISCUSSION THREAD

**Discussion by gesellix** [community/community#27505] [Actions] [answered]
With the recent change announced at https://github.blog/changelog/2022-07-19-differentiating-triggering-actor-from-executing-actor/ I'm not able to re-run dependabot pull requests as myself. I ususally have to re-run workflows, because dependabot isn't allowed to access secrets required for the workflows. Re-running them as myself enables access to the secrets. I'm aware that such a manual intervention isn't ideal by itself, but I at least I could fix failing workflows with a simple re-run.

Since the change mentioned above became effective, I don't see another way but manually changing the pull request, mangling with the commits or adding some, so that I would be the actor. This doesn't feel right, so I'd like to ask for some advice how I would easily make the workflows work. I've found https://github.com/orgs/community/discussions/5269 as a possible workaround, but I'm curious to know whether there are better solutions.

Maybe an option to choose the desired actor would be nice. By default it can be kept just as is (the initial actor), but I should be able to choose myself as actual actor for the re-run so that I could manually fall back to the previous behaviour.

**Comment by 178inaba** [comment, marked as answer]
hello!

There is a place to set the Secret for Dependabot in the Repository Settings.
If you create a Secret with the same name and value as the Actions, Dependabot can access the Secret and the job will succeed.

![スクリーンショット 2022-07-31 16 44 02](https://user-images.githubusercontent.com/12012186/182015597-e4588d02-696d-4957-b61e-adbabfca3405.png)

**Reply by yakirm-cr** [reply]
> hello!
> 
> There is a place to set the Secret for Dependabot in the Repository Settings. If you create a Secret with the same name and value as the Actions, Dependabot can access the Secret and the job will succeed.
> 
> ![スクリーンショット 2022-07-31 16 44 02](https://user-images.githubusercontent.com/12012186/182015597-e4588d02-696d-4957-b61e-adbabfca3405.png)

Does it mean that the same secret needs to be maintained twice?
If so, isn't it an overhead? (i.e. every time a secret should be updated twice, otherwise some workflows will fail when executed by one actor and succeed when executed by another)

**Reply by 178inaba** [reply]
yes. My understanding is that you need to set two secrets, one for when the user triggers the Actions, and one for the Dependabot to use when the Actions are triggered.

**Reply by gesellix** [reply]
As far as I understood, those secrets for dependabot will be available in the dependabot config, only: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-encrypted-secrets-for-dependabot
@178inaba did you give that configuration a try and made dependabot-created pull-requests work in a github actions workflow on the initial run?

**Reply by 178inaba** [reply]
yes.
Actions were failing in pull requests made by Dependabot in my repository.
But by setting Secret for Dependabot, Actions succeeded.

**Reply by 178inaba** [reply]
See this document.
https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#accessing-secrets

**Reply by gesellix** [reply]
Finally coming back to this one. You're right, thanks @178inaba for the feedback and the links to the documentation!
